src/v/storage/segment_utils.cc

```diff
@@ -163,26 +168,24 @@ ss::future<compacted_index_writer> make_compacted_index_writer(
   const std::filesystem::path& path,
   debug_sanitize_files debug,
   ss::io_priority_class iopc) {
-    return internal::make_writer_handle(path, debug)
-      .then([iopc, path](ss::file writer) {
-          try {
-              // NOTE: This try-catch is needed to not uncover the real
-              // exception during an OOM condition, since the appender allocates
-              // 1MB of memory aligned buffers
-              return ss::make_ready_future<compacted_index_writer>(
-                make_file_backed_compacted_index(
-                  path.string(),
-                  writer,
-                  iopc,
-                  segment_appender::write_behind_memory / 2));
-          } catch (...) {
-              auto e = std::current_exception();
-              vlog(stlog.error, "could not allocate compacted-index: {}", e);
-              return writer.close().then_wrapped([writer, e = e](ss::future<>) {
-                  return ss::make_exception_future<compacted_index_writer>(e);
-              });
-          }
-      });
+    try {
+        // NOTE: This try-catch is needed to not uncover the real
```

Comment:
Is this comment still correct and does anything need to be addressed here?

Before we captured the exception and returned an exceptional future containing it. Now we just rethrow after logging so I think either (a) this catch does not make sense any more since the goal seems to have been the re-wrap or (b) the catch still makes sense (i.e., the logging is important somehow, it won't be logged later?) but then the comment is wrong.

Reply:
Looking at the history, I think the original comment + try/catch was bogus.  It was giving a slightly clearer report of where/why the bad_alloc occurred in this particular case, but the current impl of spill_key_index doesn't do any big allocations on construction, so the original purpose is lost entirely.

I've removed the comment and the try/catch/log.

Reply:
Thanks John.